kokkos: add missing sycl_relocatable_device_code variant

tpadioleau opened the pull request:
Add missing Kokkos option `Kokkos_ENABLE_SYCL_RELOCATABLE_DEVICE_CODE` introduced in [Kokkos 4.5.0](https://github.com/kokkos/kokkos/blob/develop/CHANGELOG.md#sycl-1).

tpadioleau committed:
Update package.py

tpadioleau commented:
How do we handle versions older than 4.5.0 ?

cedricchevalier19 reviewed (approved):
I am not particularly fond of the name but it is coherent with cuda and hip.

We should protect variants with their minimum versions.

tpadioleau commented:
> I am not particularly fond of the name but it is coherent with cuda and hip.

Indeed i followed the existing scheme.

> We should protect variants with their minimum versions.

I can add a minimum only for this new variable.

tpadioleau committed:
Add a when clause in options_variants

tpadioleau commented:
In 7f39dabec462f2589619dde6702aa02255a1bbdd I have introduced a new entry in the `options_variants` array to describe the `when` clause in `variants`. I have moved the existing `+cuda` in the array.

I wonder if `hip_relocatable_device_code` should also have `+rocm` right to mimic `cuda_relocatable_device_code` ?

haampie commented:
Hi @tpadioleau, this package-related pull request can be migrated from `spack/spack` to `spack/spack-packages` using the [**migrate-pkg-prs utility**](https://github.com/spack/migrate-package-prs).

We encourage authors to run the migration script themselves to preserve author attribution.

You can migrate all your open pull requests at once, following the steps in the [documentation](https://github.com/spack/migrate-package-prs?tab=readme-ov-file#usage-instructions).

---

*This comment was posted automatically.*
